src/init/init.c

```diff
@@ -219,6 +220,12 @@ static int primary_core_init(int argc, char *argv[], struct sof *sof)
 	if (platform_init(sof) < 0)
 		panic(SOF_IPC_PANIC_PLATFORM);
 
+	/* Set the current value of ABI_VER in memory windows */
+	size_t abi_ver_offset = offsetof(struct ipc4_fw_registers, abi_ver);
+	uint32_t abi_ver_value = IPC4_FW_REGS_ABI_VER;
```

Comment:
shouldnt this be IPC version agnostic? What happens for people using IPC3?

Reply:
The abi version refers to IPC4 FwRegisters layout version. I changed the way the code in the init function is executed if IPC4 is supported. Thanks

Reply:
@dbaluta do we have this in IPC3 today, might be benefitial too. Fwiw, the plan with the initial register windows was to make them the same as Windows uses in IPC4, but it looks like we diverted a bit. Would be good to make this info IPC version agnostic if we can as we can put way more runtime data in there too.

Reply:
@dbaluta @lgirdwood This "slice of SRAM" is not part of the IPC3 interface and not used by the driver. The SOF Intel DSP driver does expect to have the 32bit FSR register (ROM status) in this area, and this is defined as vendor specific interface HDA_DSP_SRAM_REG_ROM_STATUS_* in sound/soc/sof/intel/hda.h , but this is not used in core SOF code on kernel side.

So I think this is a IPC4 only change at this point. If the information is useful more generally, I think we can consider extending portions (or even all) via the IPC3 ABI as well. But at least on Intel platforms this is problematic as we have some conflicts. E.g. the IPC3 build uses tracepoints that write to same area of SRAM as the IPC4 FW_REGS should be. So we can't support the exact layout in IPC3 builds.